src/Pdf.php

```diff
@@ -68,52 +73,54 @@ public function store(WithDraw $document, $fileName)
 
     /**
      * @param  string|null  $fileName
-     * @return bool
+     * @return true
      *
-     * @throws MeasureException
+     * @throws Exception
      */
-    public function withPreview($fileName = null)
+    public function withPreview(?string $fileName = null): bool
     {
-        if ($fileName && $fileName !== $this->fileName) {
-            $this->fileName = $fileName;
-        } else {
-            // Extend file name before extension
-            $extensionPos   = strrpos($this->fileName, '.');
-            $this->fileName = substr($this->fileName, 0, $extensionPos) . '_preview' . substr($this->fileName, $extensionPos);
-        }
-
-        if ($this->document instanceof WithPreview) {
-            // Make offset array key insensitive
-            $offsetArray = array_change_key_case($this->document->offset());
-
-            if (array_key_exists('x', $offsetArray)) {
-                $this->writer->setXOffset($offsetArray['x'], config('pdf.measurement.unit', 'pt'));
-            } else {
-                throw new MeasureException('No X offset defined.');
-            }
+        $mainMode = $this->previewMode;
 
-            if (array_key_exists('y', $offsetArray)) {
-                $this->writer->setYOffset($offsetArray['y'], config('pdf.measurement.unit', 'pt'));
-            } else {
-                throw new MeasureException('No Y offset defined.');
-            }
+        $this->previewMode = true;
 
-            $this->writer->enableOffset();
-        }
+        $this->previewFileName($fileName);
 
         $this->create();
 
+        // Switch back to the main mode.
+        $this->previewMode = $mainMode;
+
         return true;
     }
 
+    /**
+     * @return $this
+     */
+    public function inPreviewMode(): static
+    {
+        $this->previewMode = true;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
+    public function inOriginalMode(): static
+    {
+        $this->previewMode = false;
+
+        return $this;
+    }
+
     /**
      * Creates the pdf document(s).
      *
      * @return bool
      *
      * @throws Exception
      */
-    public function create()
+    public function create(): bool
```

Comment:
I would change the return type to `void` as we are always returning `true` anyway, in which case `boolean` is a bit misleading (you would expect that it could also be `false` if it fails but it never is).